#! /usr/bin/env python

PACKAGE='maris_pipe_pose_estimation'
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name       			Type      Level Description     		Default   Min    Max
gen.add("updateRate", double_t,    0, "update rate", 0.5, 0, 1.0)
gen.add("hitMin", int_t,    0, "minimum valid frames", 2, 0, 10)
gen.add("hitMax", int_t,    0, "hit counter saturation", 2, 0, 10)
gen.add("missMax", int_t,    0, "reset at saturation", 5, 0, 15)
gen.add("rangeOn", bool_t, 0, "check absolute distance of frame origin", False)
gen.add("rangeMin", double_t,    0, "minimum distance of frame origin [m]", 0.50, 0.0, 5.0)
gen.add("rangeMax", double_t,    0, "maximum distance of frame origin [m]", 4.00, 0.0, 8.0)
gen.add("positionOn", bool_t, 0, "check relative distance between current and previous frame origin", False)
gen.add("positionToll", double_t,    0, "relative position tollerance [m]", 2.00, 0.0, 4.0)
gen.add("orientationOn", bool_t, 0, "check relative angular displacement between current and previous frame origin", False)
gen.add("orientationToll", double_t,    0, "relative position tollerance [deg]", 45.0, 0.0, 180.0)

exit(gen.generate(PACKAGE, "maris_pipe_pose_estimation", "frame_tracker_params"))
